core/aws_ddk_core/stages/appflow_ingestion.py

```diff
@@ -161,7 +164,10 @@ def __init__(
                 resources=["*"],
             )
         )
-
+        # Additional role policy statements
+        if additional_role_policy_statements:
```

Comment:
There's more and more occurrences where we copy-paste stuff across all step function stages - e.g. alarms, policies. Makes me think that we did need that state machine stage abstraction after all

Reply:
It's fair. I am warming up to the idea of reintroducing the SFN abstraction now that we have seen a good reason for it. Up to you if you want to do it in the PR or not